src/common/utils.ts

```diff
@@ -287,3 +287,32 @@ type CamelCasedKey<S extends string> = S extends `${infer Head}_${infer Tail}`
 export type ObjectSnakeToCamelCase<T extends object> = {
   [K in keyof T as K extends string ? CamelCasedKey<K> : never]: T[K];
 };
+
+export function getMimeTypeFromArrayBuffer(arrayBuffer: ArrayBuffer) {
+  const uint8arr = new Uint8Array(arrayBuffer);
+
+  const len = 4;
+  if (uint8arr.length >= len) {
+    const signatureArr = new Array(len);
+    for (let i = 0; i < len; i++)
+      signatureArr[i] = new Uint8Array(arrayBuffer)[i].toString(16);
+    const signature = signatureArr.join('').toUpperCase();
+
+    switch (signature) {
+      case '89504E47':
+        return 'image/png';
+      case '47494638':
+        return 'image/gif';
+      case '25504446':
+        return 'application/pdf';
+      case 'FFD8FFDB':
+      case 'FFD8FFE0':
+        return 'image/jpeg';
+      case '504B0304':
+        return 'application/zip';
+      default:
+        return null;
+    }
```

Comment:
Any reference for this, was expecting we are gonna use a package?

Reply:
tried this one? https://www.npmjs.com/package/file-type

Reply:
Yes, it was the last one that made the test failing

Reply:
Can you say what happened exactly, I can help with fixing, I have no idea what the above switch case does and would rather use this package. If tests fail we can always mock the whole package as we did with others.

Reply:
our jest does not support ES modules, since file-type is ESM only

Can you confirm in the app it works for this feature and only tests break, we can then fix up tests by mocking it.